src/domains/adapter/linear/linear.adapter.ts

```diff
@@ -0,0 +1,113 @@
+      const linearError = err as LinearError;
+      throw new LinearIntegrationException(linearError.message, linearError.errors);
+    }
+  }
+
+  async integrateProjectData(linearProjectId: string, pmEmail: string): Promise<ProjectDataDTO> {
+    const linearClient = new LinearClient({
+      apiKey: process.env.LINEAR_SECRET,
+    });
+    const team: Team = await linearClient.team(linearProjectId);
+    const members: UserConnection = await team.members();
+    const stages: string[] = (await team.states()).nodes.map((stage) => stage.name);
+    const pm: User | undefined = members.nodes.find((member) => member.email === pmEmail);
+    if (pm == null) {
+      throw new ConflictException('Provided Project Manager ID not correct.');
+    }
+    let pmRole: RoleDTO | null = await this.roleRepository.getByName('Project Manager');
+    if (pmRole == null) {
+      pmRole = await this.roleRepository.create('Project Manager');
+    }
+    let dev: RoleDTO | null = await this.roleRepository.getByName('Developer');
+    if (dev == null) {
+      dev = await this.roleRepository.create('Developer');
+    }
+    let role: string;
+    const teamMembers: UserRole[] = members.nodes.map((member) => {
+      if (pmRole != null && dev != null) {
+        role = member.email === pmEmail ? pmRole.id : dev.id;
+      }
+      return new UserRole({ email: member.email, role });
+    });
+    const org: Organization = await linearClient.organization;
+
+    return new ProjectDataDTO(linearProjectId, teamMembers, team.name, stages, org.logoUrl ?? null);
+  }
+
+  async integrateAllProjectIssuesData(providerProjectId: string): Promise<IssueDataDTO[]> {
+    const linearClient = new LinearClient({
+      apiKey: process.env.LINEAR_SECRET,
+    });
+    const project: Team = await linearClient.team(providerProjectId);
+    const issues = await project.issues();
+    const integratedIssuesData: IssueDataDTO[] = [];
+    for (const issue of issues.nodes) {
+      const stage = await issue.state;
+      const creator = await issue.creator;
+      const assignee = await issue.assignee;
+      let priority: Priority;
+      switch (issue.priority) {
+        case 1:
+          priority = 'URGENT';
+          break;
+        case 2:
+          priority = 'HIGH_PRIORITY';
+          break;
+        case 3:
+          priority = 'MEDIUM_PRIORITY';
+          break;
+        case 4:
+          priority = 'LOW_PRIORITY';
+          break;
+        default:
+          priority = 'NO_PRIORITY';
+      }
+
+      integratedIssuesData.push(
+        new IssueDataDTO({
+          providerIssueId: issue.id,
+          authorEmail: creator !== undefined ? creator.email : null,
+          assigneeEmail: assignee !== undefined ? assignee.email : null,
+          providerProjectId,
+          name: issue.identifier,
+          title: issue.title,
+          description: issue.description ?? null,
+          priority,
+          storyPoints: issue.estimate ?? null,
```

Comment:
change storyPoints logic to save a string. Should check in team attribute issueEstimationType if it is estimating T-shirts size, if so, should save size (E.g. XS ) and not the number.